src/Actions/BcgovSettings.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Bcgov\Theme\Block\Actions;
+
+/**
+ * A settings page for the BCGov Block Theme, which registers the theme settings and allows users to modify them.
+ *
+ * @since 1.1.2
+ */
+class BcgovSettings {
+
+    /**
+     * Constructor.
+     */
+    public function __construct() {
+         add_action( 'admin_init', [ $this, 'bcgov_register_settings' ] );
+    }
+
+    /**
+     * Adds a top level menu to WordPress admin menu to host BCGov settings.
+     *
+     * @since 1.1.2
+     * @return void
+     */
+    public function bcgov_add_settings_menu() {
+         // Add new top level menu to host all BCGov settings.
+        add_menu_page(
+            'BCGov Settings',
+            'BCGov Block Theme Settings',
+            'manage_options',
+            'bcgov-settings-menu',
+            [ $this, 'bcgov_settings_page' ],
+            'dashicons-admin-generic',
+            77
+        );
+    }
+
+    /**
+     * Displays the BCGov Block Theme settings page.
+     *
+     * @since 1.1.2
+     * @return void
+     */
+    public function bcgov_settings_page() {
+        // Must check that the user has the required capability.
+        if ( ! current_user_can( 'manage_options' ) ) {
+            die( esc_html( __( 'You do not have sufficient permissions to access this page.' ) ) );
+        }
+
+        // Now display the settings editing screen.
+        echo '<div class="wrap">';
+
+        // Header.
+        echo '<h2>BCGov Block Theme Admin Settings</h2>';
+
+        // Settings form.
+        echo '<form name="form1" method="post" action="options.php">';
+        settings_fields( 'bcgov-settings-group' );
+        do_settings_sections( 'bcgov-theme-settings' );
+
+        echo '<p class="submit">';
+        echo '<input type="submit" name="Submit" class="button-primary" value="Save Changes" />';
```

Comment:
i feel like phpcs should have complained about this, we might want to see why it didn't.